src/acm.cpp

```diff
@@ -238,51 +238,49 @@ void ASN1_Codec::sigterm (int sig) {
-        << "(from broker "  << metadata->orig_broker_id()
-        << ":" << metadata->orig_broker_name() << std::endl;
+    std::string topics = topic.empty() ? "" : "all topics";
+    logger->info("Metadata for " + topics + "(from broker " + std::to_string(metadata->orig_broker_id()) + ":" + metadata->orig_broker_name() + ")");
 
     /* Iterate brokers */
-    std::cout << " " << metadata->brokers()->size() << " brokers:" << std::endl;
+    logger->info(" " + std::to_string(metadata->brokers()->size()) + " brokers:");
 
     for ( auto ib : *(metadata->brokers()) ) {
-        std::cout << "  broker " << ib->id() << " at " << ib->host() << ":" << ib->port() << std::endl;
+        logger->info("  broker " + std::to_string(ib->id()) + " at " + ib->host() + ":" + std::to_string(ib->port()));
     }
 
     /* Iterate topics */
-    std::cout << metadata->topics()->size() << " topics:" << std::endl;
+    logger->info(" " + std::to_string(metadata->topics()->size()) + " topics:");
 
     for ( auto& it : *(metadata->topics()) ) {
 
-        std::cout << "  topic \""<< it->topic() << "\" with " << it->partitions()->size() << " partitions:";
+        logger->info("  topic \"" + it->topic() + "\" with " + std::to_string(it->partitions()->size()) + " partitions:");
 
         if (it->err() != RdKafka::ERR_NO_ERROR) {
-            std::cout << " " << err2str(it->err());
-            if (it->err() == RdKafka::ERR_LEADER_NOT_AVAILABLE) std::cout << " (try again)";
+            logger->error("  topic \"" + it->topic() + "\" with error: " + RdKafka::err2str(it->err()));
+            if (it->err() == RdKafka::ERR_LEADER_NOT_AVAILABLE) {
+                logger->error("  topic \"" + it->topic() + "\" with error: " + RdKafka::err2str(it->err()) + " (try again)");
+            }
         }
 
-        std::cout << std::endl;
-
         /* Iterate topic's partitions */
         for ( auto& ip : *(it->partitions()) ) {
-            std::cout << "    partition " << ip->id() << ", leader " << ip->leader() << ", replicas: ";
+            logger->info("    partition " + std::to_string(ip->id()) + ", leader " + std::to_string(ip->leader()) + ", replicas: ");
 
             /* Iterate partition's replicas */
             RdKafka::PartitionMetadata::ReplicasIterator ir;
             for (ir = ip->replicas()->begin(); ir != ip->replicas()->end(); ++ir) {
-                std::cout << (ir == ip->replicas()->begin() ? "":",") << *ir;
+                logger->info(" " + std::to_string(*ir));
             }
 
             /* Iterate partition's ISRs */
-            std::cout << ", isrs: ";
+            logger->info("    partition " + std::to_string(ip->id()) + ", ISRs: ");
             RdKafka::PartitionMetadata::ISRSIterator iis;
             for (iis = ip->isrs()->begin(); iis != ip->isrs()->end() ; ++iis)
-                std::cout << (iis == ip->isrs()->begin() ? "":",") << *iis;
+                logger->info(" " + std::to_string(*iis));
```

Comment:
I don't think we need to include " " in all logging messages. In this for loop the data should be separated by something since it was being separated before.